src/compile/data/stack.ts

```diff
@@ -127,10 +127,10 @@ export class StackNode extends DataFlowNode {
       return null;
     }
 
-    let dimensionFieldDef: TypedFieldDef<string>;
+    let dimensionFieldDef: PositionFieldDef<string>;
     if (stackProperties.groupbyChannel) {
       const cDef = encoding[stackProperties.groupbyChannel];
-      dimensionFieldDef = getTypedFieldDef(cDef);
+      dimensionFieldDef = getTypedFieldDef(cDef) as PositionFieldDef<string>; // Fair to cast as groupByChannel is always either x or y
```

Comment:
If this is the case, shouldn't `encoding[stackProperties.groupbyChannel]` return a `cDef` with an appropriate type?

Reply:
Typewise cDef can be a value def / conditional field def.  However, in this case, it can't be a groupby channel if cDef isn't a field. But TS isn't smart enough to know that.  I don't think it's worth fixing here.